tls/s2n_resume.c

```diff
@@ -131,7 +131,7 @@ static int s2n_deserialize_resumption_state(struct s2n_connection *conn, struct
     return 0;
 }
 
-static int s2n_client_serialize_resumption_state(struct s2n_connection *conn, struct s2n_stuffer *to)
+int s2n_client_serialize_resumption_state(struct s2n_connection *conn, struct s2n_ticket_fields *ticket_fields, struct s2n_stuffer *to)
```

Comment:
Since we are making this function a non-static one, can we add unit tests for it?

Reply:
The changes made are minimal and involve passing a new argument to an existing, tested API. I would not block on this.

Reply:
s2n_client_serialize_resumption_state is just a wrapper around the tls12/tls13_serialize_resumption state method. It essentially just writes the encrypted ticket from the server and then calls the tls12/tls13_serialize_resumption state method. So client and server are both serializing in the same way, except the client needs to additionally add on the encrypted ticket.